src/codegen/doc/mod.rs

```diff
@@ -291,20 +290,24 @@ fn create_object_doc(w: &mut dyn Write, env: &Env, info: &analysis::object::Info
         } else {
             ty.clone()
         };
-        let fn_name = configured_functions
-            .iter()
-            .filter_map(|f| f.rename.as_ref())
-            .next()
-            .cloned();
-        create_fn_doc(w, env, function, Some(Box::new(ty)), fn_name)?;
+        if let Some(c_identifier) = &function.c_identifier {
+            // Retrieve the new_name computed during analysis, if any
+            let fn_new_name = info.functions.iter().find_map(|f| {
+                if &f.glib_name == c_identifier {
+                    f.new_name.clone()
+                } else {
+                    None
+                }
+            });
```

Comment:
I don't think you absolutely need to do the cloning inside the `find_map`, would this be more concise:

```suggestion
            let fn_new_name = info
                .functions
                .iter()
                .find(|f| &f.glib_name == c_identifier)
                .and_then(|f| f.new_name.clone());
```

Reply:
Much better, thanks! I had a feeling that we could do better (usually the case when returning `None`), but rushed to something else I guess.